src/core/server/http/router/headers.ts

```diff
@@ -37,9 +43,9 @@ export type KnownHeaders = KnownKeys<IncomingHttpHeaders>;
  * Http request headers to read.
  * @public
  */
-export type Headers = { [header in KnownHeaders]?: string | string[] | undefined } & {
+export interface Headers {
```

Comment:
Does `tsc` fail if you put the old definition of `Headers` back?

Reply:
Looks like it can be restored - pushed 47068e2a6d8ca858ea0a5d841d81530012597ba9